### PR TITLE
FUSETOOLS-2944 - added checks to prevent NPE in MavenUtils, fixes #1414

### LINKED ISSUE #1414: NullPointerException when running maven update
# New Issue Checklist

## Your Environment

- Windows 7
- Eclipse Photon (though also happened in Oxygen)
- Fuse Tooling for Eclipse 11.0.0.v20180613-2153
- JDK 1.8.0_144 as JavaSE-1.7

## Expected Behaviour

No NullPointerException occurs when running maven update.

## Actual Behaviour

After running Maven->Update Project... would get NullPointerException with the following stacktrace in the logs.

> java.lang.NullPointerException
	at org.fusesource.ide.camel.editor.utils.MavenUtils.isSyndesisExtensionProject(MavenUtils.java:275)
	at org.fusesource.ide.projecttemplates.maven.CamelProjectConfigurator.isValidCamelFacetCandidate(CamelProjectConfigurator.java:323)
	at org.fusesource.ide.projecttemplates.maven.CamelProjectConfigurator.configure(CamelProjectConfigurator.java:89)
	at org.eclipse.m2e.core.project.configurator.AbstractLifecycleMapping.configure(AbstractLifecycleMapping.java:120)
	at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager$3.call(ProjectConfigurationManager.java:508)
	at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager$3.call(ProjectConfigurationManager.java:1)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:177)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:151)
	at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager.updateProjectConfiguration(ProjectConfigurationManager.java:501)
	at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager.updateProjectConfiguration0(ProjectConfigurationManager.java:439)
	at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager$2.call(ProjectConfigurationManager.java:352)
	at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager$2.call(ProjectConfigurationManager.java:1)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:177)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:151)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:99)
	at org.eclipse.m2e.core.internal.embedder.MavenImpl.execute(MavenImpl.java:1351)
	at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager.updateProjectConfiguration(ProjectConfigurationManager.java:349)
	at org.eclipse.m2e.core.ui.internal.UpdateMavenProjectJob.runInWorkspace(UpdateMavenProjectJob.java:77)
	at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:39)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:60)

## Reproduce

It appears its because my pom (which is a child pom, not sure if that matters) does not have a build or plugins blocks. If I add empty ones, it updates fine.

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [x] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [x] Is there no duplicated code?
- [x] Are method-/class-/variable-names meaningful?

## Tests

- [x] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

File: editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java [hunk 1]
```diff
@@ -71,6 +73,7 @@ public void setup() throws CoreException {
 	}
 
 	@Test
+	@Ignore("Disabled test because its broken due to the model rework FUSETOOLS-2290")
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java]: why ignoring it? the PR is for FUSETOOLS-2944 based on commit message

File: editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java [hunk 2]
```diff
@@ -91,6 +94,7 @@ public void testUpdateMavenDependencyWithNullVersion() throws Exception {
...
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java]: why ignoring it? the PR is for FUSETOOLS-2944 based on commit message

File: editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java [hunk 3]
```diff
@@ -119,6 +123,7 @@ public void testUpdateMavenDependencyWithManagedDependency() throws Exception {
...
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java]: why ignoring it? the PR is for FUSETOOLS-2944 based on commit message

Comment (lhein) [reply editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java]: those tests are broken and ignored for a long time now. fixing them is out of scope of this PR

File: editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java (hunk 2 above)
Comment (lhein) [reply editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java]: those tests are broken and ignored for a long time now. fixing them is out of scope of this PR

File: editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java (hunk 1 above)
Comment (lhein) [reply editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java]: those tests are broken and ignored for a long time now. fixing them is out of scope of this PR

Comment (lhein) [reply editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java]: raised https://issues.jboss.org/browse/FUSETOOLS-3032 to track the repair task

Comment (apupier) [reply editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java]: ok, it was disabled at the class level before

File: editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java (hunk 2 above)
Comment (lhein) [reply editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java]: raised https://issues.jboss.org/browse/FUSETOOLS-3032 to track the repair task

File: editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java (hunk 3 above)
Comment (lhein) [reply editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java]: raised https://issues.jboss.org/browse/FUSETOOLS-3032 to track the repair task

File: editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java (hunk 2 above)
Comment (apupier) [reply editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java]: ok, it was disabled at the class level before

File: editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java (hunk 3 above)
Comment (apupier) [reply editor/tests/org.fusesource.ide.camel.editor.tests/src/test/java/org/fusesource/ide/camel/editor/utils/MavenUtilsTest.java]: ok, it was disabled at the class level before

**Comment by apupier** [comment]
remove "ready to review" as the job is red

**Comment by lhein** [comment]
testPR

**Comment by lhein** [comment]
testPR

**Comment by lhein** [comment]
testPR

**Pull request merged**